api/src/ginger_calls.rs

```diff
@@ -684,6 +684,95 @@ pub fn get_lazy_ginger_smt_path(tree: &mut LazyGingerSMT, leaf_position: u64) ->
+}
+
+pub fn flush_bvt(tree: &mut BitvectorTree) {
+    flush_ginger_smt(tree)
+}
+
+pub fn set_bvt_persistency(tree: &mut BitvectorTree, persistency: bool) {
+    set_ginger_smt_persistency(tree, persistency)
+}
+
+pub fn get_bvt_root(tree: &BitvectorTree) -> FieldElement {
+    get_ginger_smt_root(tree)
+}
+
+pub fn get_bvt_path(tree: &mut BitvectorTree, bit_position: u64) -> GingerMHTPath {
+    get_ginger_smt_path(tree, bit_position / bits_per_bvt_leaf())
+}
+
+pub fn get_bvt_leaf(tree: &BitvectorTree, bit_position: u64) -> Option<FieldElement> {
+    get_leaf_in_bvt(tree, bit_position / bits_per_bvt_leaf())
+}
+
+pub fn set_bvt_bit(tree: &mut BitvectorTree, bit_position: u64){
+    modify_bit_in_bvt(tree, bit_position, true)
+}
+
+pub fn reset_bvt_bit(tree: &mut BitvectorTree, bit_position: u64){
+    modify_bit_in_bvt(tree, bit_position, false)
+}
+
+pub fn get_bvt_bit(tree: &BitvectorTree, bit_position: u64) -> bool {
+    let leaf_bitsize = bits_per_bvt_leaf();
+
+    let leaf_index = bit_position / leaf_bitsize;
+    if let Some(leaf) = get_leaf_in_bvt(tree, leaf_index) {
+        let leaf_bits= leaf.write_bits();
+        assert_eq!(leaf_bitsize, (leaf_bits.len() - 1) as u64);
+        leaf_bits[(leaf_bitsize - (bit_position % leaf_bitsize)) as usize]
+    } else { // leaf doesn't even exist, so the bit isn't set
+        false
+    }
+}
+
+fn bits_per_bvt_leaf() -> u64 {
+    (FieldElement::size_in_bits() - 1) as u64 // using one bit lesser to make leafs never exceed the field modulus
+}
+
+fn get_leaf_in_bvt(tree: &BitvectorTree, leaf_position: u64) -> Option<FieldElement> {
+    tree.get_leaf(Coord::new(0, leaf_position as usize))
+}
+
+fn modify_bit_in_bvt(tree: &mut BitvectorTree, bit_position: u64, bit_value: bool){
+    use algebra::{Field, FromBits};
+
+    let leaf_bitsize = bits_per_bvt_leaf();
+    let leaf_index = bit_position / leaf_bitsize;
+    // println!("leaf_bitsize: {}", leaf_bitsize);
+    // println!("leaf_index: {}", leaf_index);
+    // println!("bit_position: {}", bit_position);
+    let updated_leaf = FieldElement::read_bits(
+        {
+            let mut leaf_bits =
+                if let Some(leaf) = get_leaf_in_bvt(tree, leaf_index) {
+                    leaf.write_bits()
+                } else { // leaf doesn't exist, so create an empty leaf
+                    FieldElement::zero().write_bits()
+                };
+            assert_eq!(leaf_bitsize, (leaf_bits.len() - 1) as u64);
+            // The LSB bits are at the MSB positions
+            leaf_bits[(leaf_bitsize - (bit_position % leaf_bitsize)) as usize] = bit_value;
+            leaf_bits
+        }
+    ).unwrap();
+    // println!("updated_leaf: {}", updated_leaf);
+    if !updated_leaf.is_zero() {
+        // Write an updated leaf value
+        add_leaf_to_ginger_smt(tree, &updated_leaf, leaf_index)
+    } else {
+        // Don't store empty leafs to optimize SMT size
+        remove_leaf_from_ginger_smt(tree, leaf_index)
+    }
+}
+
```

Comment:
In fact, in relation to the previous comment, how do you plan to unset all the bits of the bitvector ?

Reply:
Based on the assumption that empty leaf implies all-zero vector, it is enough to delete all leafs to unset a whole BVT